# config/example_params.yaml

---
/**:
    ros__parameters:
        general:
            grab_resolution: 'HD1200' # The native camera grab resolution. 'HD1200', 'HD2K', 'HD1080', 'HD720', 'SVGA', 'VGA', 'AUTO'
            grab_frame_rate: 60 # ZED SDK internal grabbing rate
            pub_resolution: 'NATIVE' # The resolution used for image and depth map publishing. 'NATIVE' to use the same `general.grab_resolution` - `CUSTOM` to apply the `general.pub_downscale_factor` downscale factory to reduce bandwidth in transmission
            pub_downscale_factor: 2.0 # rescale factor used to rescale image before publishing when 'pub_resolution' is 'CUSTOM'
            pub_frame_rate: 60.0 # [DYNAMIC] Frequency of publishing of visual images and depth data (not the Point Cloud, see 'depth.point_cloud_freq'). This value must be equal or less than the camera framerate.
            
        benchmark:
            tot_samples: 1000 # Total number of samples to process before exiting
            disable_dds_test: false # Disable the DDS benchmark test
            cpu_gpu_load_period: 10 # Period in milliseconds to retrieve CPU and GPU load statistics
            cpu_gpu_load_avg_wnd_size: 5 # Number of samples to use to compute the average CPU and GPU load average
            csv_log_file: 'zed_nitros_benchmark.csv' # If not empty, log the benchmark results in a CSV file with the specified name

        debug:
            debug_nitros: false # Enable Nitros debug information
            use_pub_timestamps: true # Used to measure the real communication latency. 